src/cart/cart_ops.ts

```diff
@@ -63,18 +63,21 @@ export class CartOps implements ICartOps {
         child: ItemInstance
     ): ItemInstance {
         let inserted = false;
-        const f = this.ruleChecker.getPairwiseMutualExclusionPredicate(
-            parent.key,
-            child.key
+        const f = this.ruleChecker.getIncrementalMutualExclusionPredicate(
+            parent.key
         );
         const newChildren: ItemInstance[] = [];
         for (const c of parent.children) {
-            if (f(c.key)) {
-                newChildren.push(c);
-            } else {
-                if (!inserted) {
-                    inserted = true;
-                    newChildren.push(child);
+            const existingPID = AttributeInfo.pidFromKey(c.key).toString();
+            const childPID = AttributeInfo.pidFromKey(child.key).toString();
+            if (existingPID !== childPID) {
```

Comment:
It seems that the check on line 73 should be rolled into the closure generated by getIncrementalMutualExclusionPredicate(). Do you agree?
 
If we do this, it will likely change the design of getIncrementalMutualExclusionPredicate(), since this function currently looks for identical keys, rather than identical PIDs. We should think about whether the correct behavior is to store PIDs, instead of keys.

Reply:
Will look into this and make a possible change

Reply:
So looking more into this, I don't feel confident about moving the check to `getIncrementalMutualExclusionPredicate`; we don't have access to the existing key vs the new key we want to add in there and so it makes it difficult to make that comparison. I'd like to talk through this on a call tomorrow if you're free. Maybe I'm missing something